cmd/virt-handler/virt-handler.go

```diff
@@ -73,7 +74,13 @@ func main() {
 	broadcaster.StartRecordingToSink(&kubecorev1.EventSinkImpl{Interface: coreClient.Events(api.NamespaceDefault)})
 	recorder := broadcaster.NewRecorder(kubev1.EventSource{Component: "virt-handler", Host: *host})
 
-	domainManager, err := virtwrap.NewLibvirtDomainManager(domainConn, recorder)
+	var domainManager virtwrap.DomainManager
+	switch *driver {
+	case "ovirt":
+		domainManager, err = virtwrap.NewOvirtDomainManager(domainConn, recorder)
+	default:
+		domainManager, err = virtwrap.NewLibvirtDomainManager(domainConn, recorder)
+	}
```

Comment:
Never say never.  

What this change is affecting is the calls between the shim layer and the vm launch layer.  This is not a Kubernetes API even in the existing code.  It is libvirt specific code.

What this is showing is the need for a driver architecture for the VM launch mechanism.  VDSM can be viewed as a wrapper around libvirt that provides needed functionality that cannot be expressed in libvirt.  Thus, saying that libvirt is the one-and-only-true VM launch mechanism is too harsh a stance.  

Even Docker specific code is going to be replaced by CRI at some point.

As I see it, we have two choices: drivers in tree and drivers out of tree.  Since I suspect we will have a vanishingly few number of drivers, I suggest we pull them into the tree for now.  Go seems to be built on the philosophy of static binding, and using a .so from within a go service running inside a container is going against the grain.

The first piece would be to split the existing code (looks like mostly the virtwrap file) into an interface/implementation tsetup that isolates the libvirt specific pieces.  A second commit can then implement the VDSM pieces.

Functional testing for VDSM will be problematic on a small system.  Please lay out how that would work.